templates/l_to_lgamma.cpp.in

```diff
@@ -44,27 +44,36 @@ using namespace @ModelName@_FFV_form_factors;
 
 namespace @ModelName@_l_to_lgamma {
 
-template <typename FIn, typename FOut, typename T1, typename T2>
-double lepton_total_decay_width(
-      T1 const& indices1, T2 const& indices2, 
-      const @ModelName@_mass_eigenstates& model, const softsusy::QedQcd& qedqcd) {
-
-   context_base context {model};
-
-   const auto leptonInMass = context.mass<FIn>(indices1);
-   const auto leptonOutMass = context.mass<FOut>(indices2);
-   const auto r = pow(leptonOutMass/leptonInMass, 2);
-   const auto GF2 = pow(qedqcd.displayFermiConstant(), 2);
-   const double Alpha = Sqr(FIn::electric_charge * unit_charge(context))/(4*Pi);
-
-   // eq. 10.6 of http://pdg.lbl.gov/2018/reviews/rpp2018-rev-standard-model.pdf
-   return 
-      // tree-level (with massless outgoing lepton)
-      GF2*pow(leptonInMass,5)/(192.0*pow(Pi,3)) 
-      // outgoing lepton mass correction
-      * (1 - 8*r + 8*pow(r,3) - pow(r,4) - 12*r*r*std::log(r));
-      // higher order correction
-      //* (1.0 + 0.5 * Alpha * (6.25-Sqr(Pi))/Pi)
+/**
+ * @param[in] g      Generation index of decaying lepton.
+ * @param[in] qedqcd Reference to low-energy data.
+ * @return Total decay width in GeV, according to PDG data.
+ */
+double lepton_total_decay_width(int g, const softsusy::QedQcd& qedqcd) {
+   // https://pdg.lbl.gov/2020/reviews/rpp2020-rev-phys-constants.pdf
+   constexpr double hbar = 6.582119569e-25,   // [GeV*s]
+   // https://pdg.lbl.gov/2020/listings/rpp2020-list-muon.pdf
+                    mass_muon = 0.1056583745, // [GeV]
+                    time_muon = 2.1969811e-6, // [s]
+   // https://pdg.lbl.gov/2020/listings/rpp2020-list-tau.pdf
+                    mass_tau  = 1.77686,      // [GeV]
+                    time_tau  = 290.3e-15;    // [s]
+   switch (g) {
+      case 0: throw std::invalid_argument("Electron decay is not supported");
+      case 1:
+              auto softMass = qedqcd.displayMass(softsusy::mMuon);
+              if (!is_zero(softMass - mass_muon)) {
+                 WARNING("Difference in low energy muon mass and the one of PDG");
+              }
+              return hbar/time_muon;
+      case 2:
+              auto softMass = qedqcd.displayMass(softsusy::mTau);
+              if (!is_zero(softMass - mass_tau)) {
+                 WARNING("Difference in low energy tau mass and the one of PDG");
+              }
+              return hbar/time_tau;
+      default: throw std::invalid_argument("Unrecognized lepton");
+   }
 }
```

Comment:
Sorry I have a also stupid question about this sanity check. 

 The muon and tau pole masses are inputs to the spectrum and usually there is no prediction of them in the model, but I think you keep in mind models where that is not the case because leptons may mix and we don't take account of that in the way we extract the muon yukawa.  I don't recall under what circumstances we reset `qedqcd.displayMass(softsusy::mMuon)` but does this test always work correctly for both cases?   
 
 I mean will it always pass if the pole mass is really an input, even if the user has used the flag to calculate SM masses (I am thinking of some cases where the pole mass calculation did not reverse precisely due to  the way structured the corrections)?   And in the case where there is lepton mixing, or it should get changed for some other reason, will it always get caught or it relies on the user setting the flag to calculate SM masses?

Reply:
@pathron, the idea of this, as I understand, is the following: 1) forget about all complicated things which might happen, 2) assume, that someone can somehow get mass of muon to be any number (1GeV, 10TeV - whatever), 3) warn the pure user, that something might be wrong, because of the difference. I hope, that this was exact line of thoughts of @wkotlarski .